openassessment/xblock/static/js/src/lms/oa_base.js

```diff
@@ -68,6 +69,108 @@ OpenAssessment.BaseView.prototype = {
+    srReadTexts: function(texts) {
+        var readerFeedbackID = '.sr.reader-feedback',
+            $readerFeedbackSelector = $(readerFeedbackID),
+            htmlFeedback = '';
+
+        if ($readerFeedbackSelector.length === 0) {
+            $(this.element).append(
+                '<div class="sr reader-feedback" aria-live="polite"></div>'
+            );
+            $readerFeedbackSelector = $(readerFeedbackID);
+        }
+        $.each(texts, function(ids, value) {
+            htmlFeedback = htmlFeedback + '<p>' + value + '</p>\n';
+        });
+        $readerFeedbackSelector.html(htmlFeedback);
+    },
+
+    /**
+     * Checks the rendering status of the views that may require Screen Reader Status updates.
+     *
+     * The only views that should be added here are those that require Screen Reader updates when moving from one
+     * step to another.
+     *
+     * @returns {boolean} true if any step's view is still loading.
+     */
+    areSRStepsLoading: function() {
+        return this.responseView.isRendering ||
+            this.peerView.isRendering ||
+            this.selfView.isRendering ||
+            this.gradeView.isRendering ||
+            this.trainingView.isRendering ||
+            this.staffView.isRendering;
+    },
+
+    /**
+     * Updates text int the Aria live region if all sections are rendered and focuses on the specified ID.
+     *
+     * @param {boolean} announceCurrentStatus - true if the current status should be added to the aria-live region
+     *     even if it is now showing
+     * @param {String} stepID - The id of the Step being worked on.
+     * @param {String} usageID  - The Usage id of the Step being worked on.
+     * @param {boolean} gradeStatus - true if this is a Grade status, false if it is an assessment status.
+     * @param {Object} currentView - Current active view.
+     * @param {String} focusID - The ID of the region to focus on.
+     */
+    announceStatusChangeToSRandFocus: function(announceCurrentStatus,
+                                               stepID,
+                                               usageID,
+                                               gradeStatus,
+                                               currentView,
+                                               focusID
+    ) {
+        var text = this.getStatus(stepID, usageID, gradeStatus);
+
+        if (typeof usageID !== 'undefined' &&
+            $(stepID, currentView.element).hasClass("is--showing") &&
+            typeof focusID !== 'undefined') {
+
+            $(focusID, currentView.element).focus();
+            this.srStatusUpdates.push(text);
+        } else if (announceCurrentStatus) {
+            this.srStatusUpdates.push(text);
+        }
+        if (!this.areSRStepsLoading() && this.srStatusUpdates.length > 0) {
+            this.srReadTexts(this.srStatusUpdates);
+            this.srStatusUpdates = [];
+        }
+    },
+
+    announceErrorMsgToSR: function(message) {
```

Comment:
This method is oddly named-- it has nothing to do with it being an "error message" per se. Since it is only used one place, I would just inline the code where it is called.